src/iotjs_binding.cpp

```diff
@@ -139,142 +115,123 @@ JObject::~JObject() {
 }
 
 
 JObject JObject::TypeError(const char* message) {
-  return CreateError(message, JERRY_API_ERROR_TYPE);
+  return CreateError(message, JERRY_ERROR_TYPE);
 }
 
 
 JObject JObject::URIError(const char* message) {
-  return CreateError(message, JERRY_API_ERROR_URI);
+  return CreateError(message, JERRY_ERROR_URI);
 }
 
 
 JResult JObject::Eval(const String& source,
-                      bool direct_mode,
                       bool strict_mode) {
-  JRawValueType res;
-  jerry_completion_code_t ret = jerry_api_eval(
-      reinterpret_cast<const jerry_api_char_t*>(source.data()),
+  JRawValueType res = jerry_eval(
+      reinterpret_cast<const jerry_char_t*>(source.data()),
       source.size(),
-      direct_mode,
-      strict_mode,
-      &res);
-
-  IOTJS_ASSERT(ret == JERRY_COMPLETION_CODE_OK ||
-               ret == JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION);
+      strict_mode);
 
-  JResultType type = (ret == JERRY_COMPLETION_CODE_OK)
-                     ? JRESULT_OK
-                     : JRESULT_EXCEPTION;
+  JResultType type = (jerry_value_has_error_flag(res))
+                     ? JRESULT_EXCEPTION
+                     : JRESULT_OK;
 
-  return JResult(&res, type);
+  return JResult(res, type);
 }
 
 
 void JObject::SetMethod(const char* name, JHandlerType handler) {
   IOTJS_ASSERT(IsObject());
-  JObject method(jerry_api_create_external_function(handler));
+  JObject method(jerry_create_external_function(handler));
   SetProperty(name, method);
 }
 
 
 void JObject::SetProperty(const char* name, const JObject& val) {
   IOTJS_ASSERT(IsObject());
-  JRawValueType v = val.raw_value();
-  bool is_ok  = jerry_api_set_object_field_value(
-      _obj_val.u.v_object,
-      reinterpret_cast<const jerry_api_char_t*>(name),
-      &v);
-  IOTJS_ASSERT(is_ok);
+  SetProperty(name, val.raw_value());
 }
 
 
 void JObject::SetProperty(const char* name, JRawValueType val) {
   IOTJS_ASSERT(IsObject());
-  bool is_ok  = jerry_api_set_object_field_value(
-        _obj_val.u.v_object,
-        reinterpret_cast<const jerry_api_char_t*>(name),
-        &val);
-  IOTJS_ASSERT(is_ok);
+  JRawValueType prop_name = jerry_create_string(
+      reinterpret_cast<const jerry_char_t*>(name));
+  JRawValueType ret_val = jerry_set_property(
+      _obj_val,
+      prop_name,
+      val);
+  jerry_release_value(prop_name);
+  IOTJS_ASSERT(!jerry_value_has_error_flag(ret_val));
+  jerry_release_value(ret_val);
 }
 
 
 JObject JObject::GetProperty(const char* name) {
   IOTJS_ASSERT(IsObject());
-  JRawValueType res;
-  bool is_ok = jerry_api_get_object_field_value(
-      _obj_val.u.v_object,
-      reinterpret_cast<const jerry_api_char_t*>(name),
-      &res);
-  return JObject(&res);
+  JRawValueType prop_name = jerry_create_string(
+      reinterpret_cast<const jerry_char_t*>(name));
+  JRawValueType res = jerry_get_property(_obj_val, prop_name);
+  jerry_release_value(prop_name);
+  return JObject(res);
```

Comment:
Can you create a JObject with a value with error flag? Even the original did not check is_ok. I think if there is an error we should change the ret value to undefined (by releasing the error).